beamer/tests/agent/test_request.py

```diff
@@ -377,3 +377,30 @@ def test_event_fetcher_progress_without_events(agent, direction):
                 != ape.chain.blocks[-1].number
             ):
                 sleeper.sleep(0.1)
+
+
+def test_handling_reorgs(config, direction, token, request_manager):
+    requester, target = alloc_accounts(2)
+    confirmation_blocks = 2
+    config.confirmation_blocks["l2a"] = confirmation_blocks
+    agent = Agent(config)
+    agent.start()
+
+    snap1 = ape.chain.snapshot()
+    ape.chain.mine(confirmation_blocks + 1)
+    request_id_1 = make_request(request_manager, token, requester, target, 1)
+    ape.chain.restore(snap1)
```

Comment:
Here it could happen that the agent does not get a chance to sync before we restore the chain state.
So before doing `restore`, we should wait until event fetcher's `synced_block` is `latest chain's block number - confirmation block`. Only then we should do a `restore`.